### PR TITLE
use label to set edgex name (merge to branch main)

### PR DESCRIPTION
**Pull request by theonefx**
Signed-off-by: theonefx.chenx <theonefx.chenx@alibaba-inc.com>

#### Which issue(s) this PR fixes:
Fixes https://github.com/openyurtio/yurt-device-controller/issues/47

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/clients/edgex-foundry/util.go [hunk 1]
```diff
@@ -44,20 +44,25 @@ type ClientURL struct {
 	Port int
 }
 
-func getEdgeDeviceName(d *devicev1alpha1.Device) string {
+type EdgeNameProvider interface {
```
Comment (wawlian) [inline pkg/clients/edgex-foundry/util.go:47]: Maybe you don't need to define a separate interface and use [metav1.Object](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L33) instead. The methods GetLabels() and GetName() have been defined there.

File: pkg/clients/edgex-foundry/util.go [hunk 2]
```diff
@@ -44,20 +44,25 @@ type ClientURL struct {
...
+	GetObjectMeta() metav1.Object
+	GetName() string
+}
+
+func getEdgeName(provider EdgeNameProvider) string {
```
Comment (wawlian) [inline pkg/clients/edgex-foundry/util.go:52]: Perhaps getEdgeXName is better since this util is dedicated for EdgeX Foundry?

**Review by wawlian — CHANGES_REQUESTED** [review]

File: pkg/clients/edgex-foundry/util.go (hunk 2 above)
Comment (theonefx) [reply pkg/clients/edgex-foundry/util.go:52]: good suggestion

File: pkg/clients/edgex-foundry/util.go (hunk 1 above)
Comment (theonefx) [reply pkg/clients/edgex-foundry/util.go]: done

**Review by wawlian — APPROVED** [review]

**Pull request merged**